test/server.test.js

```diff
@@ -38,6 +39,19 @@ describe('Server Creation', () => {
         await new Promise(res => server.server.on('listening', res))
         expect(info).toHaveBeenCalledWith('Serving websocket server at ws://127.0.0.1:8090. Awaiting clients...')
     })
+    it('ignores port when existing http server specified', () => {
+        const httpServer = http.createServer()
+        server = new Server({ port: 54321, server: httpServer })
+        expect(server.server._server).toBe(httpServer)
+        expect(server.server.options.port).toBe(null)
+    })
+    it('prints square brackets around ipv6 addresses when listening', async() => {
+        const info = jest.spyOn(console, 'info')
+        const httpServer = http.createServer()
+        server = new Server({ server: httpServer })
+        await new Promise(res => httpServer.listen({ port: 54322 }, res))
+        expect(info).toHaveBeenCalledWith('Serving websocket server at ws://[::]:54322. Awaiting clients...')
+    })
```

Comment:
This test is currently hanging CI because I think it's not actually closing the http server on test completion.

Reply:
I assumed WebSocketServer would close it, [but that is not the case when the server context is an options input](https://github.com/websockets/ws/blob/master/lib/websocket-server.js#L168)